lib/internal/sync/sync_routine.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/painting.dart';
+import 'package:http/http.dart';
+import 'package:potato_notes/data/dao/note_helper.dart';
+import 'package:potato_notes/data/database.dart';
+import 'package:potato_notes/data/model/sync_note.dart';
+import 'package:potato_notes/internal/preferences.dart';
+import 'package:potato_notes/internal/sync/controller/note_controller.dart';
+import 'package:potato_notes/locator.dart';
+import 'package:provider/provider.dart';
+
+class SyncRoutine {
+  List<Note> addedNotes = List();
+  List<Note> deletedNotes = List();
+  Map<Note, Map<String, dynamic>> updatedNotes = Map();
+  Preferences preferences;
+  NoteHelper noteHelper;
+
+  SyncRoutine() {
+    this.preferences = locator<Preferences>();
+    this.noteHelper = locator<NoteHelper>();
+  }
+
+  Future<bool> checkOnlineStatus() async {
+    try {
+      Response pingResponse = await get(preferences.apiUrl + "/ping");
+      if (pingResponse.statusCode != 200) return false;
+      if (pingResponse.body != "Pong!") return false;
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  Future<void> syncNotes() async {
+    if (await checkOnlineStatus() != true) {
+      return;
+    }
+    await updateLists();
+    NoteController noteController = NoteController();
```

Comment:
also this here, might want to put it in the locator

Reply:
Actually, now that i think about it, the notecontroller should just use static functions since it doesnt store state or anything

Reply:
For some reason i cant add static methods to an interface so i just added it to the locator

Reply:
can look into it, there might be another way to do so (or no way at all)

Reply:
k so you usually cant, but there really isnt a reason for keeping the abstract class, makes no sense cuz we dont have different implementations for any controller, imo you could just delete the *Interface classes and keep the controllers only with the static methods